File: 0504-base-7/solution.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        
        result = []
        
        is_negative = (True) if num < 0 else False
        num = num if num >= 0 else -1*num
        
        while num:
            remainder = num%7
            result.append(str(remainder))
            num //= 7
            
        if is_negative:
            result.append('-')
            
        return "".join(reversed(result))

File: 0504-base-7/solution.cpp
class Solution {
public:
    string convertToBase7(int num) 
    {
        if(num == 0)
            return "0";
        
        
        string result = "";
        bool is_negative = (num > 0)? false: true;
        num = (num >= 0)? num: -1*num;
        
        while(num)
        {
            result += (num%7) + '0';
            num = num/7;
        }
        if(is_negative)
            result += '-';
        
        for(int i = 0; i < result.size()/2; i++)
        {
            swap(result[i], result[result.size()-1-i]);
        }
        
        return result;
        
        
    }
};
